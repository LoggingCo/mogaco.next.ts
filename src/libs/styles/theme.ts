import { DefaultTheme } from 'styled-components';

const palette = {
  primary: {
    100: '#BFDEEE',
    200: '#9DCDE7',
    300: '#3AA5DD',
  },
  subColor: '#EBF1F4',
  fontColor: '#222222',
  fontSubColor: '#ffffff',
  error: '#EB5757',
  subError: '#FFBA52',
  gray: {
    100: '#e9e9e9',
    200: '#D9D9D9',
    300: '#656565',
  },
  yellow: '#FFE381',
  puple: '#EDC4FC',
  sky: '#A4EEFF',
} as const;

const fontSize = {
  small: '14px',
  medium: '16px',
  large: '18px',
  xLarge: '24px',
  xxLarge: '32px',
} as const;

const fontWeight = {
  thin: '100',
  regular: '400',
  medium: '500',
  bold: '700',
} as const;

const lineHeight = {
  small: '16px',
  medium: '20px',
  large: '24px',
  xLarge: '28px',
} as const;

export type PaletteTypes = typeof palette;
export type PaletteKeyTypes = keyof typeof palette;
export type FontSizeTypes = typeof fontSize;
export type FontWeightTypes = typeof fontWeight;
export type LineHeightTypes = typeof lineHeight;

const theme: DefaultTheme = {
  palette,
  fontSize,
  fontWeight,
  lineHeight,
};

export default theme;
