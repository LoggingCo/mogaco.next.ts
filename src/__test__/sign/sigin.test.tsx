import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import theme from '@/libs/styles/theme';
import SignSignUp from '@/components/Pages/sigin/SiginUp/SiginUp';

describe('<SignUp />', () => {
  it('인증완료와 아이디, 버튼이 활성화 되어야만 버튼이 활성화 된다', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <SignSignUp />
      </ThemeProvider>,
    );
    const $loginButton = screen.getByText('회원가입');
    const $authBtn = screen.getByText('인증');
  });

  // it('이메일 인증 버튼 클릭 후 해당 메일로 인증 메시지가 전송된다 ', () => {});
});
