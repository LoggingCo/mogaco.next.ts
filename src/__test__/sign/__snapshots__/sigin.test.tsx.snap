// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> enables button when both email and password are entered 1`] = `
<div>
  <div
    class="sc-kDvujY cwVVHS"
  >
    <div
      class="sc-ipEyDJ hflXJS"
    >
      <div
        class="sc-dkrFOg jcNiDM"
      >
        <a
          href="/"
        >
          mogaco
        </a>
      </div>
      <div
        class="sc-eDWCr cGxhHB"
      >
        오늘도
        <br />
        모가코,
      </div>
    </div>
    <div
      class="sc-bqWxrE iEntZY"
    >
      <div
        class="sc-ksBlkl RdDIT"
      >
        <ul>
          <li>
            로그인
          </li>
          <li>
            회원가입
          </li>
        </ul>
      </div>
      <div
        class="sc-hLBbgP FbSkc"
      >
        <form
          class="sc-eDvSVe juBFCv"
        >
          <div
            class="sc-jSUZER jhgDYF"
          >
            <input
              name="email"
              placeholder="이메일 입력하세요"
            />
            <span>
              이메일
            </span>
          </div>
          <div
            class="sc-jSUZER jhgDYF"
          >
            <input
              name="password"
              placeholder="비밀번호를 입력하세요"
              type="password"
            />
            <span>
              비밀번호
            </span>
          </div>
          <button>
            로그인
          </button>
          <div
            class="sc-gKPRtg eEOnMt"
          >
            아직 모가코 회원이 아니닌가요? 
            <span>
              모가코와 함께하기
            </span>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
